guacamole/src/main/java/org/apache/guacamole/rest/directory/DirectoryResource.java

```diff
@@ -143,13 +144,14 @@ public DirectoryResource(UserContext userContext, Directory<InternalType> direct
 
         // An admin user has access to all objects
         User self = userContext.self();
-        SystemPermissionSet systemPermissions = self.getSystemPermissions();
+        Permissions effective = self.getEffectivePermissions();
```

Comment:
Same note as above - any reason not to do

    Permissions effective = userContext.self().getEffectivePermissions()

?